runtime/signals_nat.c

```diff
@@ -57,7 +57,7 @@ void caml_garbage_collection(void)
   { /* Find the frame descriptor for the current allocation */
     d = caml_find_frame_descr(fds, retaddr);
     /* Must be an allocation frame */
-    CAMLassert(d && d->frame_size != 0xFFFF && (d->frame_size & 2));
+    CAMLassert(!FRAME_CHUNK_TOP(d) && FRAME_HAS_ALLOC(d));
```

Comment:
Before this change, `d &&` would start by checking that `d` is not NULL. It is removed from the new assertion, while `caml_find_frame_descr` may in fact return NULL. Is this intentional?

Reply:
This was a mistake. Thank you.